### PR TITLE
Improve backend

### PR DESCRIPTION
**Pull request by shiranKfir**

### CODE REVIEW

File: app/controllers/api/v1/cuisines_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+class Api::V1::CuisinesController < ApplicationController
+  include ErrorConcern
+
+  before_action :set_cuisine, only: [:show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token
+
+  # GET /cuisines
+  def index
+    @cuisines = Cuisine.all
```
Comment (shem8) [inline app/controllers/api/v1/cuisines_controller.rb]: why do you need to save it as an instance variable (`@`)?

File: app/controllers/api/v1/cuisines_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  # DELETE /cuisines/1
+  def destroy
+    @cuisine.destroy
+    head :no_content, status: :ok
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_cuisine
+      @cuisine = Cuisine.find(params[:id])
```
Comment (shem8) [inline app/controllers/api/v1/cuisines_controller.rb]: use `require`

File: app/controllers/concerns/error_concern.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module ErrorConcern
+  extend ActiveSupport::Concern
```
Comment (shem8) [inline app/controllers/concerns/error_concern.rb]: Nice

File: app/controllers/api/v1/restaurants_controller.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    head :no_content, status: :ok
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_restaurant
+      @restaurant = Restaurant.find(params[:id])
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
```
Comment (tomerdesserww) [inline app/controllers/api/v1/restaurants_controller.rb]: please set method as private

File: app/serializers/review_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class ReviewSerializer < ActiveModel::Serializer
+  attributes :id, :reviewer_name, :rating, :comment
```
Comment (tomerdesserww) [inline app/serializers/review_serializer.rb]: What about input validation? rating ,must be 0-3 number, you can go with validates or even use enumify gem

File: app/serializers/restaurant_serializer.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+class RestaurantSerializer < ActiveModel::Serializer
+  attributes :id, :title, :rating, :has_10bis, :address, :max_delivery_time
```
Comment (tomerdesserww) [inline app/serializers/restaurant_serializer.rb]: something cool that you can do here to improve performance and learn something that we use a lot here is adding an average score field and update it upon review submit with after create action

Comment (yossis) [reply app/serializers/restaurant_serializer.rb]: Its better to that under the model

**Pull request merged**